### DISCUSSION TITLE
IOL routing platform initial config

### DISCUSSION THREAD

**Discussion by DanPartelly** [ipspace/netlab#1368] [General]
Started to make experiments for IOL integration. I have the following issue: deploying the initial configs with ansible fails. "No route to host". while trying to ssh in. 

The docker containers are very fast up, ansible takes over, but it seems there is a race condition, as IOS is not yet initialized completely.

A subsequent netlab initial succeeds, and lab seems good to go. Any ideas how is this is done should be done?

**Comment by jbemmel** [comment]
See ```readiness-check``` under netsim/ansible/tasks

**Comment by ipspace** [comment]
Most VM-hosting containers have the same problem, so we have a generic `vm-clab-ssh-check.yml` task lists in the `readiness-check` directory. Many task lists are just symlinks to that one, others include that one and then do something more (for example, Nexus OS has to wait for the interfaces to become ready).

The name of your task list should be `device.yml` (where device = `iol` or whatever device code you're using) or `device-clab.yml` if you want the task list to be executed only for containers (that's the case with Nexus OS).

**Reply by DanPartelly** [reply]
Thanks both of you. I have IOL labs up and running now. Ill iron out some remaining bugs I seen towards the weekend then present to you guys as draft PR.

**Comment by ipspace** [comment]
IOL is now using readiness check.
